src/native/minipal/thread.h

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_THREAD_H
+#define HAVE_MINIPAL_THREAD_H
+
+#include <pthread.h>
+#include <stdint.h>
+#include <string.h>
+
+#if defined(__linux__)
+#include <unistd.h>
+#include <sys/syscall.h>
+#elif defined(__FreeBSD__)
+#include <pthread_np.h>
+#elif defined(__NetBSD__)
+#include <lwp.h>
+#elif defined(__HAIKU__)
+#include <kernel/OS.h>
+#endif
+
+#ifdef PTHREAD_MAX_NAMELEN_NP
+#define MINIPAL_MAX_THREAD_NAME_SIZE (PTHREAD_MAX_NAMELEN_NP - 1)
+#elif defined(__APPLE__)
+#define MINIPAL_MAX_THREAD_NAME_SIZE 63
+#elif defined(__FreeBSD__)
+#define MINIPAL_MAX_THREAD_NAME_SIZE MAXCOMLEN
+#elif defined(__HAIKU__)
+#define MINIPAL_MAX_THREAD_NAME_SIZE (B_OS_NAME_LENGTH - 1)
+#else
+#define MINIPAL_MAX_THREAD_NAME_SIZE 15
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Get the current thread ID.
+ *
+ * @return The current thread ID as a size_t value.
+ */
+static inline size_t minipal_get_current_thread_id(void)
+{
+#ifdef __wasm
+    return 0;
+#else
+    static __thread size_t tid = 0;
```

Comment:
nit1: `__thread` is a pre-C11  GCC Extension.  Maybe we should use `_Thread_local`?

nit2: there's a hidden lock here.  On Mono we need to ensure that the caller of `minipal_get_curernt_thread_id` is in GC Safe mode (preemptive suspend).  

Right now there are no mono callers, but there might be some in the future. 

Would it be more appropriate for CoreCLR & NativeAOT to cache the return value of `minipal_get_current_thread_id`?

Reply:
We could cache it in minipal once per thread to avoid contention, would that be enough? ddf5edd

Reply:
@am11 sorry, see my updated comment, I was remembering wrong where the hidden locks are. I think the way you had it before is ok

Reply:
Reverted. (and kept `_Thread_local`)